gems/sorbet-runtime/lib/types/props/private/setter_factory.rb

```diff
@@ -51,12 +53,16 @@ def self.build_setter_proc(klass, prop, rules)
           accessor_key: Symbol,
           non_nil_type: T.any(T::Types::Base, T.all(T::Props::CustomType, Module)),
           klass: T.all(Module, T::Props::ClassMethods),
+          validate: T.nilable(ValidateProc)
         )
         .returns(SetterProc)
       end
-      private_class_method def self.non_nil_proc(prop, accessor_key, non_nil_type, klass)
+      private_class_method def self.non_nil_proc(prop, accessor_key, non_nil_type, klass, validate)
         proc do |val|
           if non_nil_type.valid?(val)
+            if validate
```

Comment:
Is this `if` in the setter path OK? Or would it be better to do something like `if validate; setter_proc = then(validate, setter_proc)`?

Reply:
Does it make a difference in the `bench:setters` microbenchmark?

I'd default to making it a separate proc, but if there's no measurable difference, seems ok.

Reply:
Oh, didn't see that - cool.  No, I don't see a measurable difference between branches after a couple runs on each.